src/core_modules/capture-core/components/Pages/MainPage/WorkingLists/TemplatesLoader/TemplatesLoader.component.js

```diff
@@ -1,56 +1,79 @@
 // @flow
-import React, { memo } from 'react';
+import React, { memo, useCallback, useEffect, useRef } from 'react';
 import { withLoadingIndicator, withErrorMessageHandler } from '../../../../../HOC';
 import { TemplatesManager } from '../TemplatesManager';
 import type { Props } from './templatesLoader.types';
 
 const TemplatesManangerWithLoadingIndicator = withErrorMessageHandler()(
     withLoadingIndicator()(TemplatesManager));
 
+const useLoadTemplates = ({
+    programId,
+    loadedProgramIdForTemplates,
+    dirtyTemplates,
+    onLoadTemplates,
+    onCancelLoadTemplates,
+}) => {
+    const firstRunRef = useRef(true);
+    const triggerLoadRef = useRef(true);
+
+    let triggerLoad = false;
+    if (programId !== loadedProgramIdForTemplates ||
+        (dirtyTemplates && firstRunRef.current)) {
+        triggerLoad = true;
+    }
+    triggerLoadRef.current = triggerLoad;
+
+    const cancelLoadTemplatesIfApplicable = useCallback(() => {
+        triggerLoadRef.current && onCancelLoadTemplates && onCancelLoadTemplates();
+    }, [onCancelLoadTemplates]);
+
+    useEffect(() => {
+        firstRunRef.current = false;
+
+        if (triggerLoad) {
+            onLoadTemplates(programId);
+        }
+
+        return () => cancelLoadTemplatesIfApplicable();
+    }, [
+        triggerLoad,
+        onLoadTemplates,
+        programId,
+        cancelLoadTemplatesIfApplicable,
+    ]);
+
+    useEffect(() => () => onCancelLoadTemplates && onCancelLoadTemplates(), [onCancelLoadTemplates]);
```

Comment:
what is the difference between this "unmount" and the one above? 

What I can see and perhaps I am wrong is that, this one here is triggered each time the component is removed from the screen.  Wouldnt this cover the case that the line 38 from this file covers?

Reply:
Still cant get it. Perhaps you can explain to me in person?